dataset.py

```diff
@@ -137,3 +142,91 @@ def yearly_255_trees_partitioned(reference_csv_filename: str) -> PartitionedData
   print_split(partition_data)
   return partition_data
 
+#Utility function for randomly sampling a point around a sample site
+def is_valid_point(lat: float, lon: float, reference_isocape: raster.AmazonGeoTiff):
+  return True if raster.get_data_at_coords(reference_isocape, lon, lat, 0) else False
+
+# Pick a random point around (lat, lon) within max_distance_km. If edge_only is
+# true, only pick points exactly max_distance_km away from (lat, lon).
+def random_nearby_point(lat, lon, max_distance_km, edge_only=False):
+  # Pick a random angle pointing outward from the origin.
+  # 0 == North, 90 == East, 180 == South, 270 == West
+  angle = 360 * random.random()
+
+  # sqrt() is required for an equal radial distribution, otherwise samples
+  # cluster around origin.
+  dist = max_distance_km if edge_only else max_distance_km * math.sqrt(random.random())
+
+  # WGS-84 is the most accurate ellipsoidal model of Earth, but we should double
+  # check to make sure this matches the model used by our sample collectors.
+  point = distance.geodesic(
+      ellipsoid='WGS-84', kilometers=dist).destination((lat, lon), bearing=angle)
+  return point.latitude, point.longitude
+
+# Given a list of real_points, returns true if (lat, lon) is within threshold
+# of any of those points.
+def is_nearby_real_point(lat, lon, real_points, threshold_km):
+  for point, _ in real_points:
+    if distance.geodesic((lat, lon), point).km < threshold_km:
+      return True
+  return False
+
+
+#This function creates a dataset based on real samples adding a Fraud column
+def create_fraudulent_samples(real_samples_data, mean_iso,element,max_trusted_radius,max_fraud_radius,min_fraud_radius):
```

Comment:
From the functions you added, I think this is the only one we want to surface as a public API. Could we replace the "helper" functions to have an underscore such that we follow the standard to name them like that if we don't intend them to use them beyond being helper functions? i.e. _is_nearby_real_point

Reply:
Also, please add function parameter data types.

Reply:
Per offline conversation, please add dataset_test.ipynb where we test only create_fraudulent_samples.